src/EllipsisNotation.jl

```diff
@@ -3,16 +3,45 @@ __precompile__()
 module EllipsisNotation
 
   import Base.getindex, Base.setindex!
-  const ..   = Val{:..}
+  const .. = Val{:..}()
+  const IndexType = Union{(VERSION < v"0.6-" ? Real : Int), Val{:..}}
```

Comment:
The reason for https://github.com/JuliaLang/julia/commit/8a55211f3b9c20ea2baedabbecd56ff9f4286284 was to enable things like this without worrying about what methods specific AbstractArray subtypes have defined.  You should now overload `to_indices` and it'll *just work* for all array types and all indexing functions (get/setindex and views).  Let me know if you have any questions there.